mlrun/utils/helpers.py

```diff
@@ -944,19 +944,22 @@ def fill_object_hash(object_dict, uid_property_name, tag=""):
     return uid
 
 
-def fill_artifact_object_hash(
-    object_dict, uid_property_name, iteration=None, producer_id=None
-):
+def fill_artifact_object_hash(object_dict, iteration=None, producer_id=None):
```

Comment:
you can extract the artifact specific attributes needed (db_key, tree, etc.) and then reuse `fill_object_hash` above.